### PR TITLE
util/stringutil: migrate test-infra to testify

### LINKED ISSUE #26421: migrate test-infra to testify for `util/stringutil` pkg

### PR DESCRIPTION
**Pull request by feitian124**
### What problem does this PR solve?

Issue Number: close #26421 

### Release note

- No release note

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by disksing — APPROVED** [review]

File: util/stringutil/string_util_test.go [hunk 1]
```diff
@@ -65,20 +54,19 @@ func (s *testStringUtilSuite) TestUnquote(c *C) {
...
-		x, err := Unquote(t.str)
-		c.Assert(x, Equals, t.expect)
-		comment := Commentf("source %v", t.str)
-		if t.ok {
-			c.Assert(err, IsNil, comment)
+	for _, v := range table {
+		x, err := Unquote(v.str)
+		require.Equal(t, v.expect, x)
+		if v.ok {
+			require.Nil(t, err)
```
Comment (xhebox) [inline util/stringutil/string_util_test.go:61]: There is `Nilf`, and `NotNilf`, the alternative for `Commentf("source %v", t.str)`, ditto for other comments. Comments are useful for debugging.

Comment (feitian124) [reply util/stringutil/string_util_test.go:61]: ok thanks, i will ehance it

**Review by xhebox — APPROVED** [review]

File: util/stringutil/string_util_test.go (hunk 1 above)
Comment (feitian124) [reply util/stringutil/string_util_test.go]: hi, i finished the enhance, please review again.
i searched whole code repo and it seems that i am the first one to use `Nilf` and `NotNilf` function.
i manually made some test fail, seems the testfiy output is clean enough and no `Nilf` related comment is show on the idea terminal. not sure this is expected or not.
```shell
/usr/local/go/bin/go tool test2json -t /tmp/___TestUnquote_in_github_com_pingcap_tidb_util_stringutil.test -test.v -test.paniconexit0 -test.run ^\QTestUnquote\E$
=== RUN   TestUnquote
=== PAUSE TestUnquote
=== CONT  TestUnquote
    string_util_test.go:59: 
        	Error Trace:	string_util_test.go:59
        	Error:      	Not equal: 
        	            	expected: "abcdef"
        	            	actual  : "123abcdef"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-abcdef
        	            	+123abcdef
        	Test:       	TestUnquote
--- FAIL: TestUnquote (0.00s)
```

Comment (tisonkun) [reply util/stringutil/string_util_test.go]: I'd suggest you write "source %v" to cover the case where v=""

**Comment by disksing** [comment]
CI failed. Not sure if it is related.

File: util/stringutil/string_util_test.go (hunk 1 above)
Comment (feitian124) [reply util/stringutil/string_util_test.go]: get it

**Comment by tisonkun** [comment]
```
[2021-07-26T08:42:06.291Z]     testing.go:1092: race detected during execution of test
[2021-07-26T08:42:06.291Z] --- FAIL: TestT (15.79s)
[2021-07-26T08:42:06.291Z] === CONT  
[2021-07-26T08:42:06.291Z]     testing.go:1092: race detected during execution of test
[2021-07-26T08:42:06.291Z] FAIL
[2021-07-26T08:42:06.291Z] FAIL	github.com/pingcap/tidb/table/tables	16.026s
```

I think it is unrelated. But we should avoid any test failure...

We should reopen https://github.com/pingcap/tidb/issues/25691

**Comment by disksing** [comment]
Thanks for your contribution @feitian124 . I'll summon a bot to merge your PR.

**Pull request merged**